File: app/src/main/java/com/example/montijo/explorerquest_android/CharacterSelectionActivity.java
package com.example.montijo.explorerquest_android;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

public class CharacterSelectionActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_character_selection);

        // find Dora button
        Button dora = (Button) findViewById(R.id.button_dora);

        // set onClickListener to Dora Button
        dora.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // call character creation for Dora
                createCharacter("Dora");
            }
        });

        // find Boots button
        Button boots = (Button) findViewById(R.id.button_boots);

        // set onClickListener to Boots Button
        boots.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // call character creation for Boots
                createCharacter("Boots");
            }
        });
    }

    private void createCharacter (String character) {
        // create chosen character
        Character hero = new Character(character);

        // create intent to start Story activity
        Intent storyIntent = new Intent(CharacterSelectionActivity.this, StoryActivity.class);

        // pass character to activity intent
        storyIntent.putExtra("Hero", hero);

        // start the new activity
        startActivity(storyIntent);
    }
}

File: app/src/main/java/com/example/montijo/explorerquest_android/CharacterSelectionActivity.kt
package com.example.montijo.explorerquest_android

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class CharacterSelectionActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_character_selection)

        // find Dora button
        val dora = findViewById<View>(R.id.button_dora) as Button

        // set onClickListener to Dora Button
        dora.setOnClickListener { // call character creation for Dora
            createCharacter("Dora")
        }

        // find Boots button
        val boots = findViewById<View>(R.id.button_boots) as Button

        // set onClickListener to Boots Button
        boots.setOnClickListener { // call character creation for Boots
            createCharacter("Boots")
        }
    }

    private fun createCharacter(character: String) {
        // create chosen character
        val hero = Character(character)

        // create intent to start Story activity
        val storyIntent = Intent(this@CharacterSelectionActivity, StoryActivity::class.java)

        // pass character to activity intent
        storyIntent.putExtra("Hero", hero)

        // start the new activity
        startActivity(storyIntent)
    }
}